### PR TITLE
Fix compilation with msys2 (cmake edition)

### PR DESCRIPTION
**Pull request by SmileTheory**
So let's try this again.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,10 @@ prboom2/release
 /prboom2/VisualC2015/ReleaseServer/prboom_server.tlog
 /prboom2/VisualC2015/ReleaseServer
 /build*
+CMakeFiles
+CMakeCache.txt
+Makefile
+*.cmake
+config.h
+prboom-plus.wad
+*.exe
```
Comment (alexey-lysiuk) [inline .gitignore]: Please use out-of-tree builds. The `build*` directories filter was added especially for this.

File: prboom2/src/CMakeLists.txt [hunk 2]
```diff
@@ -312,6 +312,8 @@ target_include_directories(prboom-plus PRIVATE
 target_link_libraries(prboom-plus PRIVATE
     ${OPENGL_LIBRARIES}
     ${SDL2_LIBRARIES}
+    winmm
+    comctl32
```
Comment (alexey-lysiuk) [inline prboom2/src/CMakeLists.txt]: This is not the right place for Windows specific libraries. As you can see, it broke Linux and macOS targets.

Comment (SmileTheory) [reply prboom2/src/CMakeLists.txt]: https://github.com/coelckers/prboom-plus/pull/37/commits/780c682cc2de4056a71816f4958cfe78dec47ff5#diff-ec649d8d689ac310801f57cae7a3f446R316

File: .gitignore (hunk 1 above)
Comment (SmileTheory) [reply .gitignore]: https://github.com/coelckers/prboom-plus/pull/37/commits/780c682cc2de4056a71816f4958cfe78dec47ff5#diff-a084b794bc0759e7a6b77810e01874f2L41

**Pull request merged**